node/hyperbahn-client.js

```diff
@@ -380,10 +380,8 @@ function unadvertise(opts) {
     timers.clearTimeout(self._advertisementTimer);
     timers.clearTimeout(self.advertisementTimeoutTimer);
     self.latestAdvertisementResult = null;
-    self.state = States.UNADVERTISED;
-    self.emit('unadvertised');
     function unadvertiseInternalCb(error, result) {
-        if (error) {
+        if (error && error.type !== 'tchannel.connection.reset') {
```

Comment:
when hyperbahn backend receive unad request, it will close and delete the peer, which causes all connections of the peer to close. And if connection reset error response comes back earlier (sometimes but not always) than unad success response, client side will see this error. Draining unad request will turn success response into declined error, which again show up as an error at client side.